core/src/test/java/org/apache/accumulo/core/client/admin/TabletMergeabilityInfoTest.java

```diff
@@ -0,0 +1,102 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+public class TabletMergeabilityInfoTest {
+
+  @Test
+  public void testValidation() {
+    // ALWAYS requires an insertion time
+    assertThrows(IllegalArgumentException.class,
+        () -> new TabletMergeabilityInfo(TabletMergeability.always(), Optional.empty(),
+            () -> Duration.ofDays(10)));
+
+    // NEVER requires empty insertion time
+    assertThrows(IllegalArgumentException.class,
+        () -> new TabletMergeabilityInfo(TabletMergeability.never(), Optional.of(Duration.ZERO),
+            () -> Duration.ofDays(10)));
+
+    // Delay can't be null
+    assertThrows(IllegalArgumentException.class,
+        () -> new TabletMergeabilityInfo(TabletMergeability.after(Duration.ofMillis(-10)),
+            Optional.of(Duration.ofHours(1)), () -> Duration.ofDays(10)));
+
+    // InsertionTime can't be negative
+    assertThrows(IllegalArgumentException.class,
+        () -> new TabletMergeabilityInfo(TabletMergeability.always(),
+            Optional.of(Duration.ofHours(-1)), () -> Duration.ofDays(10)));
+
+    // No negative supplier for current time
+    assertThrows(NullPointerException.class,
+        () -> new TabletMergeabilityInfo(TabletMergeability.always(),
+            Optional.of(Duration.ofHours(1)), null));
+  }
+
+  @Test
+  public void testNever() {
+    var tmi = new TabletMergeabilityInfo(TabletMergeability.never(), Optional.empty(),
+        () -> Duration.ofDays(10));
+    assertFalse(tmi.isMergeable());
+    assertTrue(tmi.getTabletMergeability().isNever());
+    assertTrue(tmi.getElapsed().isEmpty());
+    assertTrue(tmi.getDelay().isEmpty());
+    assertTrue(tmi.getRemaining().isEmpty());
+  }
+
+  @Test
+  public void testAlways() {
+    var tmi = new TabletMergeabilityInfo(TabletMergeability.always(),
+        Optional.of(Duration.ofDays(1)), () -> Duration.ofDays(10));
+    assertTrue(tmi.isMergeable());
+    assertTrue(tmi.getTabletMergeability().isAlways());
+    assertTrue(tmi.getElapsed().orElseThrow().toNanos() > 0);
+    assertEquals(Duration.ZERO, tmi.getRemaining().orElseThrow());
+  }
+
+  @Test
+  public void testDelay() {
+    // test values when isMergeable() is false
+    var tmi = new TabletMergeabilityInfo(TabletMergeability.after(Duration.ofDays(2)),
+        Optional.of(Duration.ofDays(9)), () -> Duration.ofDays(10));
+    assertFalse(tmi.isMergeable());
+    assertEquals(Duration.ofDays(2), tmi.getDelay().orElseThrow());
+    assertEquals(Duration.ofDays(1), tmi.getElapsed().orElseThrow());
+    assertEquals(tmi.getDelay().map(delay -> delay.minus(tmi.getElapsed().orElseThrow())),
+        tmi.getRemaining());
+
```

Comment:
This test inseration time being after current time.

```suggestion
    // test insertion time after current steady time
    var tmi3 = new TabletMergeabilityInfo(TabletMergeability.after(Duration.ofDays(2)),
            Optional.of(Duration.ofDays(11)), () -> Duration.ofDays(10));
    assertFalse(tmi3.isMergeable());
    assertEquals(Duration.ofDays(2), tmi3.getDelay().orElseThrow());
    assertEquals(Duration.ZERO, tmi3.getElapsed().orElseThrow());
    assertEquals(Duration.ofDays(2), tmi3.getRemaining().orElseThrow());
```

Reply:
This is a good idea to add a test for this